mkosi

```diff
@@ -2017,10 +2017,13 @@ def make_squashfs(args, workspace, for_cache):
     if for_cache:
         return None
 
-    comp_args = [] if args.compress is True else ['-comp', args.compress]
+    comp_args = args.mksquashfs_extra_args if args.mksquashfs_extra_args else ['-noappend']
+    if args.compress is not True:
+        assert args.compress is not False
+        comp_args += ['-comp', args.compress]
     with complete_step('Creating squashfs file system'):
         f = tempfile.NamedTemporaryFile(dir=os.path.dirname(args.output), prefix=".mkosi-squashfs")
-        run(["mksquashfs", os.path.join(workspace, "root"), f.name, "-noappend", *comp_args],
+        run(["mksquashfs", os.path.join(workspace, "root"), f.name, *comp_args],
```

Comment:
hmm, I am kinda afraid that this might end up in an inflation of similar options sooner or later for all the other tools, and hence I'd prefer if we'd find a generic solution for this so that we can cover other tools with a similar approach. 

here's one idea: maybe, let's change run() to redirect the executed command line to `./mkosi.tool-<command>` if that file exists and is executable. i.e. if you want to replace or tweak the "mksquashfs" command you'd drop a shell script in called `./mkosi.tool-mksquashfs", and it would just exec the real mksquashfs but add in some args. Such a scheme could eventually be useful for allow other, less established distros to be hooked in, i.e. if "./mkosi.tool-pkg-install" or so exists we'd use that tool instead of apt-get/dnf/...